repository/src/test/java/org/apache/atlas/services/PurgeServiceTest.java

```
@@ -0,0 +1,186 @@
+        Iterator<AtlasIndexQuery.Result> iterator = Arrays.asList(result).iterator();
+        when(mockIndexQuery.vertices(anyInt(), anyInt())).thenReturn(iterator);
+
+        // mock entityStore to expand deletion candidates to include the same vertex
+        Set<AtlasVertex> deletionCandidates = new HashSet<>();
+        deletionCandidates.add(vertex);
+        when(entityStore.accumulateDeletionCandidates(any())).thenReturn(deletionCandidates);
+
+        // mock WorkItemsQualifier and its results queue
+        PurgeService.WorkItemsQualifier wiq = mock(PurgeService.WorkItemsQualifier.class);
+        Queue results = new ArrayDeque();
+        AtlasEntityHeader header = new AtlasEntityHeader();
+        header.setGuid("guid-1");
+        header.setTypeName("typeA");
+        results.add(header);
+        when(wiq.getResults()).thenReturn(results);
+        doNothing().when(wiq).shutdown();
+        doNothing().when(wiq).checkProduce(any());
+
+        // spy createQualifier and getEntityTypes
+        doReturn(new HashSet<>(Arrays.asList("typeA"))).when(purgeService).getEntityTypes();
+        doReturn(wiq).when(purgeService).createQualifier(any(), any(), any(), anyInt(), anyInt(), Mockito.eq(true));
+
+        EntityMutationResponse response = purgeService.purgeEntities();
+
+        Assert.assertNotNull(response);
+        Assert.assertNotNull(response.getPurgedEntities());
+        Assert.assertEquals(response.getPurgedEntities().size(), 1);
+
+        // verify next-level calls
+        verify(purgeService, times(1)).createQualifier(any(), any(), any(), anyInt(), anyInt(), Mockito.eq(true));
+        verify(entityStore, times(1)).accumulateDeletionCandidates(any());
+
+        // verify that checkProduce got called at least once with a vertex
+        ArgumentCaptor<AtlasVertex> vertexCaptor = ArgumentCaptor.forClass(AtlasVertex.class);
+        verify(wiq, times(1)).checkProduce(vertexCaptor.capture());
```

Why are we checking if checkProduce is getting called here?

---

It ensures future changes don’t silently drop the critical enqueue step. Without this verify, someone could remove or bypass wiq.checkProduce(vertex) and the test might still pass (because the queue was prefilled), hiding a real bug. By asserting the call and its exact argument, the test fails immediately if enqueueing is removed, reordered, or altered.